chainer/sequential.py

```diff
@@ -0,0 +1,383 @@
+
+    def __radd__(self, other):
+        if callable(other):
+            return Sequential(other) + self
+        else:
+            raise ValueError('add (+) operator is support only callable '
+                             'objects, but {} is not callable'.format(
+                                 str(type(other))))
+
+    def __iadd__(self, other):
+        if isinstance(other, Sequential):
+            for layer in other:
+                self.append(layer)
+        elif callable(other):
+            self.append(other)
+        else:
+            raise ValueError('add (+) operator is support only callable '
+                             'objects, but {} is not callable'.format(
+                                 str(type(other))))
+        return self
+
+    def __mul__(self, n_repeat):
+        if n_repeat <= 0:
+            return Sequential()
+        ret = self.copy()
+        for _ in range(n_repeat - 1):
+            for layer in self:
+                if isinstance(layer, chainer.Link):
+                    layer = copy.deepcopy(layer)
+                    for param in layer.params(include_uninit=False):
+                        param.initialize(param.shape)
+                else:
+                    layer = copy.copy(layer)
+                ret.append(layer)
+        return ret
+
+    def __rmul__(self, n_repeat):
+        return self * n_repeat
+
+    def __imul__(self, n_repeat):
+        if n_repeat <= 0:
+            self.clear()
+            return self
+        n_layers = len(self)
+        for _ in range(n_repeat - 1):
+            for i in range(n_layers):
+                if isinstance(self[i], chainer.Link):
+                    layer = copy.deepcopy(self[i])
+                    for param in layer.params(include_uninit=False):
+                        param.initialize(param.shape)
+                else:
+                    layer = copy.copy(layer)
+                self.append(layer)
+        return self
+
+    def __call__(self, *x):
+        """Forward pass computation.
+
+        This method performs the forward pass computation by giving the input
+        variable ``x`` to the layers registered in the constructor in the same
+        order as the order in which the argments are given to the constructor.
+
+        It should be noted that the input variable is given directly to the
+        first layer and all intermediate outputs generated during the forward
+        pass are also directly feeded to the next layer. Therefore, the number
+        of output at a layer should be the same as the number of input at the
+        next layer.
+
+        Args:
+            x: Input variables.
+
+        Returns:
+            The output of the final layer in the given layers.
+
+        """
+        for layer in self._layers:
+            if isinstance(x, tuple):
+                x = layer(*x)
+            else:
+                x = layer(x)
+        return x
+
+    def __reduce__(self):
+        if self._n_lambda > 0:
```

Comment:
Is `_n_lambda` only used here? If so, I feel it's better to count the number of lambda functions on the fly here.

Reply:
Yes, it's used only here. So that's right. I'll change to calculate it here.